### ISSUE TITLE
Implement regularized loss functions (L1, L2, Elastic Net)

### ISSUE THREAD

**Issue by nktr-cp** [nktr-cp/multilayer-perceptron#34]
## 概要
過学習を防ぐために、L1正則化とL2正則化を含む正則化付き損失関数を実装する必要があります。これらの正則化手法により、モデルの複雑さを制御し、より汎化性能の高いモデルを構築できます。

## 背景
現在の損失関数は基本的なバイナリクロスエントロピーのみをサポートしていますが、実用的なニューラルネットワークでは重みの大きさを制限する正則化項を追加することが一般的です。L1正則化はスパースな重みを促進し、L2正則化は重みの分散を抑制します。

## 要件
- [ ] L1正則化（Lasso正則化）の実装
  - 重みの絶対値の和に基づくペナルティ
  - スパース性の促進
- [ ] L2正則化（Ridge正則化）の実装
  - 重みの二乗和に基づくペナルティ
  - 重みの分散抑制
- [ ] Elastic Net正則化（L1 + L2の組み合わせ）
- [ ] 既存の損失関数への正則化項の統合
- [ ] 設定可能な正則化強度パラメータ（λ）
- [ ] 層ごとの個別正則化設定オプション

## 技術詳細
- `src/loss.rs` モジュールの拡張
- 新しい構造体: `RegularizedLoss`
- 正則化項の計算とグラデーション
- ネットワーク全体の重みアクセス機構
- 設定可能な正則化パラメータ

### L1正則化
- R_L1(W) = λ₁ * Σ|wᵢⱼ|
- ∂R_L1/∂wᵢⱼ = λ₁ * sign(wᵢⱼ)

### L2正則化
- R_L2(W) = λ₂ * Σwᵢⱼ²
- ∂R_L2/∂wᵢⱼ = 2λ₂ * wᵢⱼ

### 正則化付き損失
- L_total = L_original + R_L1(W) + R_L2(W)

## 設定例
```rust
// L2正則化付きバイナリクロスエントロピー
let loss = RegularizedLoss::new(
    BinaryCrossEntropy::new(),
    RegularizationConfig {
        l1_lambda: 0.0,
        l2_lambda: 0.001,
        apply_to_bias: false,
    }
);

// Elastic Net正則化
let loss = RegularizedLoss::new(
    BinaryCrossEntropy::new(),
    RegularizationConfig {
        l1_lambda: 0.001,
        l2_lambda: 0.001,
        apply_to_bias: false,
    }
);
```

## 実装オプション
- バイアス項への正則化適用/非適用の選択
- 層ごとの異なる正則化強度
- 動的正則化強度の調整
- 正則化項の可視化

## 受け入れ基準
- L1、L2、Elastic Net正則化が正しく実装されている
- 既存の損失関数と統合されている
- 正則化強度を柔軟に設定できる
- バイアス項への適用を制御できる
- 正則化の効果を検証するテストが追加されている
- 過学習抑制効果のベンチマークが含まれている
- ドキュメントと使用例が提供されている
